java-checks/src/main/java/org/sonar/java/checks/MembersDifferOnlyByCapitalizationCheck.java

```diff
@@ -108,8 +109,8 @@ private static boolean isValidIssueLocation(Symbol currentMember, Symbol knownMe
     return !sameOwner(currentMember, knownMember) || isOverriding(knownMember) || getDeclarationLine(currentMember) > getDeclarationLine(knownMember);
   }
 
-  private static boolean bothPublic(Symbol s1, Symbol s2) {
-    return s1.isPublic() && s2.isPublic();
+  private static boolean sameVisibilityNotPrivate(Symbol s1, Symbol s2) {
+    return !s1.isPrivate() && ((JavaSymbol) s1).flags() == ((JavaSymbol) s2).flags();
```

Comment:
This is a buggy as flags can include final modifier for instance. Please fix it.